starport/services/serve/serve.go

```diff
@@ -302,39 +389,194 @@ func (s *Serve) initSteps(ctx context.Context, conf starportconf.Config) (
-			key      = &bytes.Buffer{}
-			mnemonic = &bytes.Buffer{}
+		steps.Add(s.createAccountSteps(ctx, account.Name, account.Coins, false)...)
+	}
+
+	for _, execOption := range s.plugin.ConfigCommands() {
+		steps.Add(step.New(step.NewOptions().
+			Add(execOption).
+			Add(s.stdSteps(logAppcli)...)...,
+		))
+	}
+
+	steps.Add(step.New(step.NewOptions().
+		Add(s.plugin.GentxCommand(conf)).
+		Add(s.stdSteps(logAppd)...)...,
+	))
+	steps.Add(step.New(step.NewOptions().
+		Add(step.Exec(
+			s.app.d(),
+			"collect-gentxs",
+		)).
+		Add(s.stdSteps(logAppd)...)...,
+	))
+	return
+}
+
+func (s *Serve) relayerSteps(ctx context.Context, sconf *starportsecretconf.Config) (steps step.Steps) {
+	selfAcc, found := sconf.SelfRelayerAccount(s.app.n())
+	if !found {
+		return nil
+	}
+
+	// prep relayer config
+	rlyConf := rly.Config{
+		Global: rly.GlobalConfig{
+			Timeout:       "10s",
+			LiteCacheSize: 20,
+		},
+		Paths: rly.Paths{},
+	}
+	rpcAddress, err := s.rpcAddress()
+	if err != nil {
+		// TODO no panic
+		panic(err)
+	}
+
+	rlyConf.Chains = append(rlyConf.Chains, rly.NewChain(selfAcc.Name, rpcAddress))
+	for _, acc := range sconf.Relayer.Accounts {
+		rlyConf.Chains = append(rlyConf.Chains, rly.NewChain(acc.ID, acc.RPCAddress))
+		rlyConf.Paths[fmt.Sprintf("%s-%s", selfAcc.Name, acc.ID)] = rly.NewPath(
+			rly.NewPathEnd(selfAcc.Name, acc.ID),
+			rly.NewPathEnd(acc.ID, selfAcc.Name),
 		)
+	}
+	home, err := os.UserHomeDir()
+	if err != nil {
+		// TODO no panic
+		panic(err)
+	}
+	relayerHome := filepath.Join(home, s.app.nd(), "relayer")
+	if err := os.MkdirAll(filepath.Join(relayerHome, "config"), os.ModePerm); err != nil {
+		// TODO no panic
+		panic(err)
+	}
+
+	configPath := filepath.Join(relayerHome, "config/config.yaml")
+	file, err := os.OpenFile(configPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+	if err != nil {
+		// TODO no panic
+		panic(err)
+	}
+	defer file.Close()
+	if err := yaml.NewEncoder(file).Encode(rlyConf); err != nil {
+		// TODO no panic
+		panic(err)
+	}
+
+	var (
+		key = &bytes.Buffer{}
+	)
+	for _, acc := range sconf.Accounts {
+		acc := acc
+		steps.Add(step.New(
+			step.Exec(
+				s.app.d(),
+				"keys",
+				"add",
+				acc.Name,
+				"--recover",
+				"--keyring-backend", "test",
+			),
+			step.Write([]byte(acc.Mnemonic+"\n")),
+		))
 		steps.Add(step.New(step.NewOptions().
 			Add(
-				s.plugin.AddUserCommand(account.Name),
+				s.plugin.ShowAccountCommand(acc.Name),
+				step.PostExec(func(err error) error {
+					if err != nil {
+						return err
+					}
+					coins := strings.Join(acc.Coins, ",")
+					key := strings.TrimSpace(key.String())
+					return cmdrunner.
+						New().
+						Run(ctx, step.New(step.NewOptions().
+							Add(step.Exec(
+								s.app.d(),
+								"add-genesis-account",
+								key,
+								coins,
+							)).
+							Add(s.stdSteps(logAppd)...)...,
+						))
+				}),
+			).
+			Add(step.Stdout(key))...,
+		))
+		fmt.Println(relayerHome, selfAcc.Name)
+		steps.Add(step.New(
```

Comment:
Same as above, we should give a sentence to describe what does each step, or redirect the developer to the relayer doc

Reply:
done by #336 and 9f4ad8c